api/src/db/schema.ts

```diff
@@ -0,0 +1,141 @@
+    googleId: text().notNull(),
+    name: text().notNull(),
+    email: text().notNull(),
+    picture: text().notNull(),
+    theme: text(),
+    lastRoadmapEditAt: timestamp(),
+  },
+  (table) => ({
+    uniqueGoogleId: unique('unique_google_id').on(table.googleId),
+  }),
+);
+
+export const report = pgTable(
+  'report',
+  {
+    id: integer().primaryKey().generatedAlwaysAsIdentity(),
+    reviewId: integer()
+      .notNull()
+      .references(() => review.id, { onDelete: 'cascade' }),
+    reason: text().notNull(),
+    createdAt: timestamp().defaultNow().notNull(),
+  },
+  (table) => ({
+    reviewIdIdx: index('reports_review_id_idx').on(table.reviewId),
+  }),
+);
+
+export const review = pgTable(
+  'review',
+  {
+    id: integer().primaryKey().generatedAlwaysAsIdentity(),
+    professorId: text().notNull(),
+    courseId: text().notNull(),
+    userId: integer()
+      .notNull()
+      .references(() => user.id),
+    anonymous: boolean().notNull(),
+    content: text(),
+    rating: integer().notNull(),
+    difficulty: integer().notNull(),
+    gradeReceived: text().notNull(),
+    createdAt: timestamp().defaultNow().notNull(),
+    updatedAt: timestamp().defaultNow(),
+    forCredit: boolean().notNull(),
+    quarter: text().notNull(),
+    takeAgain: boolean().notNull(),
+    textbook: boolean().notNull(),
+    attendance: boolean().notNull(),
+    tags: text().array(),
+    verified: boolean().notNull().default(false),
+  },
+  (table) => ({
+    ratingCheck: check('rating_check', sql`${table.rating} >= 1 AND ${table.rating} <= 5`),
+    difficultyCheck: check('difficulty_check', sql`${table.difficulty} >= 1 AND ${table.difficulty} <= 5`),
+    unique: unique('unique_review').on(table.userId, table.professorId, table.courseId),
+    professorIdIdx: index('reviews_professor_id_idx').on(table.professorId),
+    courseIdIdx: index('reviews_course_id_idx').on(table.courseId),
+  }),
+);
+
+export const planner = pgTable(
+  'planner',
+  {
+    id: integer().primaryKey().generatedAlwaysAsIdentity(),
+    userId: integer().references(() => user.id),
+    name: text().notNull(),
+    years: jsonb().$type<SavedPlannerYearData>().array().notNull(),
+  },
+  (table) => ({
+    userIdIdx: index('planners_user_id_idx').on(table.userId),
+  }),
+);
+
+export const transferredCourse = pgTable(
+  'transferred_course',
+  {
+    userId: integer().references(() => user.id),
+    courseName: text(),
+    units: real(),
+  },
+  (table) => ({
+    userIdIdx: index('transferred_courses_user_id_idx').on(table.userId),
+  }),
+);
+
+export const vote = pgTable(
+  'vote',
+  {
+    reviewId: integer()
+      .notNull()
+      .references(() => review.id, { onDelete: 'cascade' }),
+    userId: integer()
+      .notNull()
+      .references(() => user.id),
+    vote: integer().notNull(),
+  },
+  (table) => ({
+    voteCheck: check('votes_vote_check', sql`${table.vote} = 1 OR ${table.vote} = -1`),
```

Comment:
is there a reason for using `sql'...'` here instead of something like `or()`?

Reply:
I think `or()` works here? I have to use a non-null assertion for some reason but I think it's working.
```ts
    voteCheck: check('votes_vote_check', or(eq(table.vote, 1), eq(table.vote, -1))!),
```

Reply:
Great, I think that's better for consistency

Reply:
for future reference, `or()` does not work here. we have to use `sql` for checks